lib/context.js

```diff
@@ -16,59 +16,74 @@
 
 const rclnodejs = require('bindings')('rclnodejs');
 
-let defaultContext = undefined;
+let defaultContext = null;
 
 /**
- * @class - Class representing a Context in ROS
+ * Class representing a Context in ROS
  * @hideconstructor
  */
-
 class Context {
   constructor() {
     this._handle = rclnodejs.createContext();
   }
 
   /**
-   * Shutdown the context.
+   * Shut down the context.
    * @return {undefined}
+   * @throws {Error} If there is a problem shutting down the context.
    */
   shutdown() {
-    rclnodejs.shutdown(this._handle);
+    rclnodejs.shutdown(this.handle);
+    if (this.isDefaultContext) {
+      defaultContext = null;
+    }
   }
 
   /**
-   * Try to shutdown the context.
+   * Try to shut down the context.
    * @return {undefined}
+   * @throws {Error} If there is a problem shutting down the context.
    */
   tryShutdown() {
-    if (rclnodejs.ok(this._handle)) {
-      rclnodejs.shutdown(this._handle);
+    if (this.isOk) {
+      this.shutdown();
     }
   }
 
-  handle() {
+  /**
+   * Get the handle referencing the internal context object. Do not modify it yourself: only pass it to *rclnodejs* functions!
+   * @return {undefined} a reference to the internal context object
+   */
+  get handle() {
```

Comment:
Please change from `context.hand()` to `context.handle` in index.js because of getter.

Reply:
Done in 84d7816